opencog/query/PatternMatch.cc

```diff
@@ -35,8 +35,7 @@ PatternMatch::PatternMatch(void) : _used(false) {}
 /// This is just a convenience wrapper around do_match().
 void PatternMatch::match(PatternMatchCallback *cb,
                          Handle hvarbles,
-                         Handle hclauses,
-                         Handle hnegates)
```

Comment:
Ooof. The "negates" are critical for handling optional clauses.  There are cases where you want to find patterns of a certain shape, but only if they do not have a certain sub-shape.  Alternately, you want to do a different kind of handling, if the subshape is present. ...  I am hoping that some other changes elsewhere restore this functionality ...